libensemble/gen_classes/aposmm.py

```diff
@@ -0,0 +1,123 @@
+import copy
+from typing import List
+
+import numpy as np
+from numpy import typing as npt
+
+from libensemble.generators import LibensembleGenThreadInterfacer
+from libensemble.message_numbers import EVAL_GEN_TAG, PERSIS_STOP
+from libensemble.tools import add_unique_random_streams
+
+
+class APOSMM(LibensembleGenThreadInterfacer):
+    """
+    Standalone object-oriented APOSMM generator
+    """
+
+    def __init__(
+        self, History: npt.NDArray = [], persis_info: dict = {}, gen_specs: dict = {}, libE_info: dict = {}, **kwargs
+    ) -> None:
+        from libensemble.gen_funcs.persistent_aposmm import aposmm
+
+        gen_specs["gen_f"] = aposmm
+        if not gen_specs.get("out"):  # gen_specs never especially changes for aposmm even as the problem varies
+            n = len(kwargs["lb"]) or len(kwargs["ub"])
+            gen_specs["out"] = [
+                ("x", float, n),
+                ("x_on_cube", float, n),
+                ("sim_id", int),
+                ("local_min", bool),
+                ("local_pt", bool),
+            ]
+            gen_specs["persis_in"] = ["x", "f", "local_pt", "sim_id", "sim_ended", "x_on_cube", "local_min"]
+        if not persis_info:
+            persis_info = add_unique_random_streams({}, 2, seed=4321)[1]
+        super().__init__(History, persis_info, gen_specs, libE_info, **kwargs)
+        if not self.persis_info.get("nworkers"):
+            self.persis_info["nworkers"] = gen_specs["user"]["max_active_runs"]  # ??????????
+        self.all_local_minima = []
+        self._ask_idx = 0
+        self._last_ask = None
+        self._tell_buf = None
+        self._n_buffd_results = 0
+        self._n_total_results = 0
+        self._told_initial_sample = False
+
+    def _slot_in_data(self, results):
+        """Slot in libE_calc_in and trial data into corresponding array fields. *Initial sample only!!*"""
```

Comment:
this is what you want:
`self._tell_buf[self._n_buffd_results:self._n_buffd_results + len(results)] = results`